### ISSUE TITLE
Array to string exception caused by Variants field with default value

### ISSUE THREAD

**Issue by jmauzyk** [verbb/formie#1498]
### Describe the bug

Getting the following exception when attempting to save a form with a variants field that has a default value:
```
2023-07-13 14:49:45 [web.ERROR] [yii\base\ErrorException:2] yii\base\ErrorException: Array to string conversion in /var/www/html/vendor/craftcms/cms/src/helpers/Db.php:604
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(79): yii\base\ErrorHandler->handleError(2, 'Array to string...', '/var/www/html/v...', 604)
#1 /var/www/html/vendor/craftcms/cms/src/helpers/Db.php(604): craft\web\ErrorHandler->handleError(2, 'Array to string...', '/var/www/html/v...', 604)
#2 /var/www/html/vendor/craftcms/cms/src/helpers/Db.php(855): craft\helpers\Db::parseParam('elements.id', Array, '=', false, 'integer')
#3 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1408): craft\helpers\Db::parseNumericParam('elements.id', Array)
#4 /var/www/html/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\pgsql\QueryBuilder))
#5 /var/www/html/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build(Object(craft\commerce\elements\db\VariantQuery))
#6 /var/www/html/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Query->createCommand(Object(craft\db\Connection))
#7 /var/www/html/vendor/craftcms/cms/src/db/Query.php(247): yii\db\Query->all(NULL)
#8 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1567): craft\db\Query->all(NULL)
#9 /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php(350): craft\elements\db\ElementQuery->all()
#10 /var/www/html/vendor/verbb/formie/src/base/FormFieldTrait.php(214): verbb\formie\fields\formfields\Variants->defineValueAsString(Object(craft\commerce\elements\db\VariantQuery), Object(verbb\formie\elements\Submission))
#11 /var/www/html/vendor/verbb/formie/src/helpers/Variables.php(405): verbb\formie\fields\formfields\Variants->getValueAsString(Object(craft\commerce\elements\db\VariantQuery), Object(verbb\formie\elements\Submission))
#12 /var/www/html/vendor/verbb/formie/src/helpers/Variables.php(377): verbb\formie\helpers\Variables::_getParsedFieldValue(Object(verbb\formie\fields\formfields\Variants), Object(craft\commerce\elements\db\VariantQuery), Object(verbb\formie\elements\Submission), NULL)
#13 /var/www/html/vendor/verbb/formie/src/helpers/Variables.php(255): verbb\formie\helpers\Variables::_getParsedFieldValues(Object(verbb\formie\elements\Form), Object(verbb\formie\elements\Submission), NULL)
#14 /var/www/html/vendor/verbb/formie/src/controllers/SubmissionsController.php(1179): verbb\formie\helpers\Variables::getParsedValue('{formName} \xE2\x80\x94\xC2...', Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#15 /var/www/html/vendor/verbb/formie/src/controllers/SubmissionsController.php(1152): verbb\formie\controllers\SubmissionsController->_setTitle(Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#16 /var/www/html/vendor/verbb/formie/src/controllers/SubmissionsController.php(465): verbb\formie\controllers\SubmissionsController->_populateSubmission(Object(verbb\formie\elements\Form))
#17 [internal function]: verbb\formie\controllers\SubmissionsController->actionSubmit()
#18 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#19 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#20 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('submit', Array)
#21 /var/www/html/vendor/craftcms/cms/src/web/Application.php(304): yii\base\Module->runAction('formie/submissi...', Array)
#22 /var/www/html/vendor/craftcms/cms/src/web/Application.php(607): craft\web\Application->runAction('formie/submissi...', Array)
#23 /var/www/html/vendor/craftcms/cms/src/web/Application.php(283): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#24 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#25 /var/www/html/web/en-us/index.php(12): yii\base\Application->run()
#26 {main} {"memory":9287344,"exception":"[object] (yii\\base\\ErrorException(code: 2): Array to string conversion at /var/www/html/vendor/craftcms/cms/src/helpers/Db.php:604)"} 
```

### Steps to reproduce

1. Create a form with a variants field and set a default value for the field.
2. Fill out the form, leaving the variants field unfilled so it falls back to its default value.
3. Submit the form.

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.4.15

### Plugin version

2.0.33

### Multi-site?

Yes

**Comment by engram-design** [comment]
I can't seem to replicate this, sorry. By the way, when you say `Fill out the form, leaving the variants field unfilled so it falls back to its default value.` that should be impossible because the value of the dropdown (if you have it set to a dropdown) automatically selects the value you set in the field settings. If you remove that to pick the "none" option, then that's what the form will save upon submission.

**Comment by jmauzyk** [comment]
Sorry, I forgot to mention a key detail, which is that the variants field has Visibility set to Disabled for the front end. That's how it's possible to submit without making a selection for that field. My intention is to have this field available to team members on the CP for internal use, but to default to a certain variant when the user submits. Not sure if is the intended use case for this sort of field and settings.

**Comment by engram-design** [comment]
That's definitely a valid use case, and I see what you mean - that I can replicate.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.33"`.

**Comment by engram-design** [comment]
Fixed in [2.0.34](https://github.com/verbb/formie/releases/tag/2.0.34)
